framework/graphics/dx12_util.cpp

```diff
@@ -992,9 +992,10 @@ uint64_t GetAvailableGpuAdapterMemory(IDXGIAdapter3* adapter, const bool is_uma)
         }
         if (SUCCEEDED(adapter->QueryVideoMemoryInfo(0, memory_segment, &video_memory_info)))
         {
-            if (video_memory_info.Budget > video_memory_info.CurrentUsage)
+            uint64_t total_memory = video_memory_info.Budget * memory_usage;
```

Comment:
The `memory_usage` argument is not clamped so `total_memory` could be > `video_memory_info.Budget`. Is it allowed to for an application to use more than `video_memory_info.Budget`? If `memory_usage < 0`, what should `total_memory` be? This should also use a static_cast from double to uint64_t.